### PR TITLE
SQL: Update configuration pages

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
this pr completes the changes for updating SQL datasource configuration pages.
this can be split into three parts:
- psql config page update
- small mysql update to keep consistency with psql
- shared tls component update

**Special notes for your reviewer:**

**PSQL Before & After:**
<div>
Screenshot 2023-09-27 at 11 50 48
Screenshot 2023-09-27 at 11 49 18
</div>

---

**TLS Before & After:**
<div>
Screenshot 2023-09-27 at 11 51 00
Screenshot 2023-09-27 at 11 49 59
</div>

### CODE REVIEW

File: public/app/features/plugins/sql/components/configuration/TLSSecretsConfig.tsx [hunk 1]
```diff
@@ -23,60 +24,94 @@ export const TLSSecretsConfig = <T extends DataSourceJsonData, S extends {} = {}
...
+                </Tooltip>
+              </Stack>
+            </Label>
           }
-          labelWidth={labelWidth}
-          label="TLS/SSL Client Certificate"
         >
-          <SecretTextArea
-            placeholder="Begins with -----BEGIN CERTIFICATE-----"
+          <SecretInput
```
Comment (aangelisc) [inline public/app/features/plugins/sql/components/configuration/TLSSecretsConfig.tsx:45]: Do we still want this to be a larger text field?

**Review by aangelisc — APPROVED** [review]
Just one nit in the comments on the TLS fields. LGTM great work!

File: public/app/features/plugins/sql/components/configuration/TLSSecretsConfig.tsx (hunk 1 above)
Comment (gwdawson) [reply public/app/features/plugins/sql/components/configuration/TLSSecretsConfig.tsx:45]: thats a fair point, i feel like there was a reason i changed it but i'm not sure what that was 😆. ill change it back.

File: public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx [hunk 2]
```diff
@@ -163,69 +156,76 @@ export const ConfigurationEditor = (props: DataSourcePluginOptionsEditorProps<My
...
+          </Field>
+        </ConfigSubSection>
+
+        <ConnectionLimits options={options} onOptionsChange={onOptionsChange} />
       </ConfigSection>
 
-      <Divider />
-
-      <ConnectionLimits options={options} onOptionsChange={onOptionsChange} />
+      {config.secureSocksDSProxyEnabled && (
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx:223]: Shouldn't this be under additional settings?

File: public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx [hunk 3]
```diff
@@ -163,69 +156,76 @@ export const ConfigurationEditor = (props: DataSourcePluginOptionsEditorProps<My
       <Divider />
 
       <ConfigSection title="Additional settings">
```
Comment (zoltanbedi) [inline public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx:158]: In the document it says that additional settings should be collapsable.

**Review by zoltanbedi — COMMENTED** [review]
DataSourceDescription hasRequiredFields should be true. The host url is required.

**Review by zoltanbedi — COMMENTED** [review]
In the Github data source the info box was changed to a collapsible section. Is this going to stay like this?
Screenshot 2023-10-10 at 15 47 31

File: public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx [hunk 4]
```diff
@@ -73,12 +73,8 @@ export const ConfigurationEditor = (props: DataSourcePluginOptionsEditorProps<My
             onChange={onDSOptionChanged('url')}
           />
         </Field>
-      </ConfigSection>
-
-      <Divider />
 
-      <ConfigSection title="Authentication">
-        <Field label="Database name">
+        <Field label="Database name" required>
```
[the remark this answers is not in the document]
Comment (gwdawson) [reply public/app/plugins/datasource/mysql/configuration/ConfigurationEditor.tsx:77]: i didn't know this, thanks for pointing it out!

**Comment by gwdawson** [comment]
**New user permissions:**
Screenshot 2023-10-16 at 10 24 13

**Review by asimpson — APPROVED** [review]

**Review by zoltanbedi — CHANGES_REQUESTED** [review]
@gwdawson We are almost there. Can you please change the user permission alert in the mysql config as well?

**Review by zoltanbedi — APPROVED** [review]
Thank you! LGTM

**Pull request merged**